### PR TITLE
added router transitions

### PR DESCRIPTION
**Pull request by culinovic-a**

### CODE REVIEW

File: src/app/app.component.ts [hunk 1]
```diff
@@ -1,10 +1,12 @@
 import { Component, OnInit } from '@angular/core';
 import { OverlayContainer } from '@angular/cdk/overlay';
+import { routerAnimation } from './shared/animations/router.animation';
```
Comment (tabuckner) [inline src/app/app.component.ts:3]: YES! Great job :D

File: src/app/app.component.ts [hunk 2]
```diff
@@ -14,4 +16,8 @@ export class AppComponent implements OnInit {
   ngOnInit() {
     this.overlayContainer.getContainerElement().classList.add('light-theme');
   }
+
+  public getRouterOutletState(outlet) {
+    return outlet.isActivated ? outlet.activatedRoute : '';
```
Comment (tabuckner) [inline src/app/app.component.ts:21]: just want to show you something:

image

Look familiar? :P 

Did the same thing on my enterprise application at work :D

File: src/app/shared/animations/router.animation.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  transition,
+  style,
+  query,
+} from '@angular/animations';
+
+export const routerAnimation = trigger('routerAnimation', [
+  transition('* => *', [
+    query(':enter', [style({ opacity: 0 })], { optional: true }),
+
+    query(
```
Comment (tabuckner) [inline src/app/shared/animations/router.animation.ts:13]: if you're still not getting the behavior you'd like with this animation, take a look at this example. The animation is similar:

```typescript
export const routeFadeIn = trigger('routeFadeIn', [
  transition('* => *', [
    query(':leave', style({ position: 'absolute', left: 0, right: 0, opacity: 1, marginLeft: '24px', marginRight: '24px' }), { optional: true }),
    query(':enter', style({ position: 'absolute', left: 0, right: 0, opacity: 0, marginLeft: '24px', marginRight: '24px' }), { optional: true }),
    query(':leave', animate('.1s', style({ opacity: 0 })), { optional: true }),
    query(':enter', animate('.1s', style({ opacity: 1 })), { optional: true })
  ])
]);
```

I think the only real difference is setting the `style` on the `:leave` and `:enter` queries separate from the `animate`.

Try that approach and see if you get any closer to your desired behavior.

File: src/app/app.component.ts (hunk 2 above)
Comment (culinovic-a) [reply src/app/app.component.ts:21]: > image
> 
> Look familiar? :P
> 

Aaaaa, great minds think alike! :D

**Pull request merged**